Feature/collateral to uint256

cityhunteur opened the pull request:
TODO

- [x] Fix `TestMarginUpdateOnOrder`
- [x] Fix `TestMTMLossSocialization`

Commit:
refactor(collateral): migrate to uint256

jeremyletang reviewed (commented):
OK, I started reviewing mostly the engine.go file.

A couple of comments:
- when returning an account or a balance we should clone the actual bigUint
- When passing a uint as parameter, eventually cloning too if needed.
- Also let's try to avoid unecessary allocation when possible for the equivalent of a += b.

Commit:
refactor(events)!: update LossSocialization amount with loss and adj

BREAKING CHANGE: refactor this event so it supports both loss and
adjustment amounts. This requires an update to the corresponding proto

Commit:
refactor(events)!: revert change to LossSocialization event

EVODelavega committed:
Rounding using num types

EVODelavega committed:
cast to bytes, remove test

Commit:
refactor(collateral): resolve pr issues

Commit:
refactor(collateral): resolve pr issues

EVODelavega committed:
Clean up collateral

EVODelavega committed:
Merge branch 'feature/collateral-to-uint256' of github.com:vegaprotocol/vega into feature/collateral-to-uint256

 Conflicts:
	collateral/engine.go

EVODelavega committed:
Ensure AssetSource can only be ERC20 or BuiltIn

EVODelavega committed:
Hash full uint256

EVODelavega committed:
Update test

EVODelavega committed:
Set margin shortfall + ensure min amount is set

EVODelavega committed:
Clone amounts when getting transfer requests

EVODelavega committed:
Tweak tests

jeremyletang committed:
remove unecessary files (these are only on develop for now)

EVODelavega committed:
CR comments
